src/components/SearchResultsPage/SearchResultsPage.jsx

```diff
@@ -9,27 +9,34 @@ import dataArray from "../../data";
 
 const SearchResultsPage = () => {
   const [films, setFilms] = useState(dataArray);
-  const [isFilmActive, setIsFilmActive] = useState(false);
-  const [activeFilm, setActiveFilm] = useState();
-  const len = films[films.length - 1].id;

```

Comment:
instead of passing films to addModal we can create handler - function (handleSetFilm ) that will calculate new id, add it to new film and than call setFilms